xbmc/cores/VideoPlayer/DVDCodecs/Video/VAAPI.cpp

```diff
@@ -123,14 +128,44 @@ bool CVAAPIContext::EnsureContext(CVAAPIContext **ctx, CDecoder *decoder)
   return true;
 }
 
+namespace
+{
+int find_open_render_node_fd()
```

Comment:
It really seems I don't see it. As in my opinion only the EGLDisplay matters. As this one decides at the end ...

But to shorten this discussion that leads to nowhere. @BrandonSchaefer if you want please add a mir backend - it only needs to do what the android backend does. For the vaPutSurface you can just return UNIMPLEMENT - vaapi has such an error return value.

Then this can be sent to the VAAPI guys for review. If they say: that's totally not needed, just use drm prime - fine. if they take it, also fine.

We - in kodi need that drm mode anyways the very moment we speak transcoding, so it will come back in a different form anyways. Especially for the use case when we use a display and additionally use the kodi instance for transcoding - we will directly open the DRM no matter what display server is running for us.